### PR TITLE
Add concept map documentation

### PR DESCRIPTION
**Pull request by neenjaw**
Recent comments on the forum bring to light that there is relatively sparse documentation on:
- the requirements that were determined for the concept map
- the method of computation determining the specification for the concept map
- details about the resulting graph

### CODE REVIEW

File: building/tracks/concept-map.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Concept Map
+
+The concept map is one of the main methods to illustrate the progression that a student takes through the concept exercises.
+
+## Design Goals
+
+- Provide a meaningful concept map of ideas from simple to complex ideas.
+- Provide a pathway towards fluency in a language.
+- Illustrate the progression through the course [track] contents.
```
Comment (ErikSchierboom) [inline building/tracks/concept-map.md]: What does `[track]` refer to?

File: building/tracks/concept-map.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+```mermaid
+flowchart TD
+    Basics --> Integers
+    Integers --> Strings
+    Basics --Edge Removed--> Strings
+
+    linkStyle 2 stroke-width:2px,fill:none,stroke:red;
+```
+
+[github-concept-code]: https://github.com/exercism/website/blob/e47525f66b036662138e5e41055430fe7963a282/app/commands/track/determine_concept_map_layout.rb
```
Comment (ErikSchierboom) [inline building/tracks/concept-map.md]: Maybe point to the version that is in `main`?

```suggestion
[github-concept-code]: https://github.com/exercism/website/blob/main/app/commands/track/determine_concept_map_layout.rb
```

File: building/tracks/concept-map.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    - Complete: Concepts whose learning exercise is complete, but one or more practice exercise is not complete.
+    - Available: Concepts whose learning exercise has not been completed.
+    - Locked: Concepts whose learning exercise requires one or more pre-requisite exercise to be completed.
+- Paths
+  - Provide a relation of one concept to the next.
+  - Provide a relation indicating the building blocks of a concept back to the root.
+
+## Configuration
+
+The configuration of the concept map is determined by details of the `config.json` associated to a track.
```
Comment (ErikSchierboom) [inline building/tracks/concept-map.md]:
```suggestion
The configuration of the concept map is determined by details of the [`config.json`](/docs/building/tracks/config-json) associated to a track.
```

But then as a reference link

**Review by ErikSchierboom — COMMENTED** [review]
Couple of tiny comments:

- Could you add an entry to https://github.com/exercism/docs/blob/main/building/config.json
- Could you refer to the document from the "Concepts" section here: https://github.com/exercism/docs/blob/main/building/tracks/README.md#concepts
- Maybe also add a link to this document from https://github.com/exercism/docs/blob/main/building/tracks/concepts.md?

File: building/tracks/concept-map.md (hunk 1 above)
Comment (neenjaw) [reply building/tracks/concept-map.md]: I had added that as a colloquialism, but probably isn't necessary

Comment (neenjaw) [reply building/tracks/concept-map.md]:
```suggestion
- Illustrate the progression through the course contents.
```

**Comment by ErikSchierboom** [comment]
Great. All that's left to do is:

> Could you add an entry to https://github.com/exercism/docs/blob/main/building/config.json
> Could you refer to the document from the "Concepts" section here: https://github.com/exercism/docs/blob/main/building/tracks/README.md#concepts
> Maybe also add a link to this document from https://github.com/exercism/docs/blob/main/building/tracks/concepts.md?

**Comment by neenjaw** [comment]
@ErikSchierboom Yea, I didn't have time to address those points yet.  Will re-request review when completed. 😄

**Comment by neenjaw** [comment]
I think I have those other areas covered now, @ErikSchierboom 😄

**Review by ErikSchierboom — APPROVED** [review]
Excellent!

**Pull request merged**